internal/services/currency/currencyService.go

```diff
@@ -2,31 +2,28 @@ package cs
 
 import (
 	"context"
-	"errors"
-	"net/http"
+	"fmt"
 	"subscription-api/internal/entities"
+	ss "subscription-api/internal/services"
 )
 
-type ConvertParams struct {
-	From entities.Currency
-	To   []entities.Currency
+type CurrencyAPIClient interface {
+	Convert(ctx context.Context, baseCcy entities.Currency, targetCcies []entities.Currency) (map[entities.Currency]float64, error)
 }
-
-type CurrencyService interface {
-	Convert(ctx context.Context, params ConvertParams) (map[entities.Currency]float64, error)
-}
-
-var InvalidArgumentErr = errors.New("invalid argument")
-var FailedPreconditionErr = errors.New("failed precondition")
-var InvalidRequestErr = errors.New("invalid-request")
-
 type currencyService struct {
```

Comment:
Seems this service just proxing request to apiClient. Do we really need this service?

Reply:
In perspective I want to provide saving currency rates to db and add method that allows another daemon to update 
rates automatically. It is all about decreasing amount of made requests to third party API.

Reply:
Gotcha